### DISCUSSION TITLE
LFS Client Error 405

### DISCUSSION THREAD

**Discussion by willRiVR** [git-lfs/git-lfs#5190] [Q&A] [answered]
We have lfs installed on a gitlab instance running on a synology docker instance. As far as I can tell I have everything configured, am just pushing a single png and get this error:

LFS: Client error: http://gitlab.test/group/project.git/gitlab-lfs/objects/5917ac1f70be397f2f3bfcfd4a2215017df74700f221c8739256f2a39e9805a5/5963 from HTTP 405

have an lfsconfig file with this in:

[lfs]
	url = http://gitlab.test:8080/group/project.git/info/lfs

thanks in advance for any help!

**Comment by bk2204** [comment, marked as answer]
Hey,

Usually a 405 means that the method in question (`GET`, `POST`, or `PUT`) is not allowed.  Can you attempt the same operation with `GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1` prepended to it and provide the output here as an attachment?

My guess is what's going wrong here is that your service actually runs on port 8080, but you haven't told your GitLab server that.  The server has to know what port it's running on because the batch request provides absolute URLs to upload and download files, and it looks like right now it's suggesting you use the default HTTP port (80), which probably means that the `PUT` request is going to the wrong service, which is why you're getting the 405.

If that's the case, then you'll need to configure the GitLab instance correctly to provide the expected URL so that this works.

**Reply by willRiVR** [reply]
[output.txt](https://github.com/git-lfs/git-lfs/files/10024736/output.txt)

thanks for the reply! Here is the output, it seemed to just retry this over and over. I can push non lfs repos absolutely fine, so would that suggest it's not a port issue? Or do I need to set that specifically for lfs as well?

**Reply by bk2204** [reply]
Yup, that confirms it.  The Git protocol doesn't make multiple requests, so it won't have this problem.  You're probably seeing retries because the Git LFS client also retries, which the Git client does not.

If your GitLab server is using port 8080, then you definitely need to configure that in GitLab for LFS to work here.  [This answer on ServerFault](https://serverfault.com/questions/585528/set-gitlab-external-web-port-number) worked at least at some point in the past, I presume.

**Reply by willRiVR** [reply]
Apologies for not updating this sooner, thank you for the assistance as the problem is now fixed! Issue was as you said, gitlab within the docker container believed it was on 80, while the actual docker port was 8080. This worked fine until lfs requested the url from gitlab and got back the 80 address. We changed the port to 8080 on the container and set it correctly as the external url on gitlab, and it now all works. Only other gotcha was we had to set the puma port to 8081 to stop a collision.

thanks again!
